src/widgets/Dialogs.js

```diff
@@ -312,6 +316,8 @@ define(function (require, exports, module) {
 
         // Pipe dialog-closing notification back to client code
         $dlg.one("hidden", function () {
+            BrambleEvents.triggerDialogClosed();
```

Comment:
I think you should move this down inside the `setTimeout` below, where `result.resolve` happens.